driver-core/src/test/unit/com/mongodb/internal/connection/DefaultConnectionPoolSpecification.groovy

```diff
@@ -381,11 +431,15 @@ class DefaultConnectionPoolSpecification extends Specification {
 
     def selectConnectionAsync(DefaultConnectionPool pool) {
         def serverLatch = new ConnectionLatch()
-        pool.getAsync { InternalConnection result, MongoException e ->
-            serverLatch.connection = result
-            serverLatch.throwable = e
-            serverLatch.latch.countDown()
+        SingleResultCallback<InternalConnection> callback = new SingleResultCallback<InternalConnection>() {
```

Comment:
I do not understand how/why passing `groovy.lang.Closure` to `pool.getAsync` instead of `SingleResultCallback` was working. However, as soon as I introduced the stubbing [here|https://github.com/mongodb/mongo-java-driver/pull/685/files#diff-67ac2cdea14dc7b5e3520f0dead251b92af6648d8384ab077dc56d20dda6e9a1R321], I started getting the following exception:
```java
ERROR org.mongodb.driver.connection - Callback onResult call produced an error
groovy.lang.MissingMethodException: No signature of method: com.mongodb.internal.connection.DefaultConnectionPoolSpecification$_selectConnectionAsync_closure1.doCall() is applicable for argument types: (null, UncheckedIOException) values: [null, java.io.UncheckedIOException: expected failure]
Possible solutions: doCall(com.mongodb.internal.connection.InternalConnection, com.mongodb.MongoException), findAll(), findAll()
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:255) ~[groovy-2.5.12.jar:2.5.12]
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1041) ~[groovy-2.5.12.jar:2.5.12]
	at groovy.lang.Closure.call(Closure.java:405) ~[groovy-2.5.12.jar:2.5.12]
	at org.codehaus.groovy.runtime.ConvertedClosure.invokeCustom(ConvertedClosure.java:50) ~[groovy-2.5.12.jar:2.5.12]
	at org.codehaus.groovy.runtime.ConversionHandler.invoke(ConversionHandler.java:122) ~[groovy-2.5.12.jar:2.5.12]
	at com.sun.proxy.$Proxy37.onResult(Unknown Source) ~[na:na]
	at com.mongodb.internal.async.ErrorHandlingResultCallback.onResult(ErrorHandlingResultCallback.java:48) ~[main/:na]
	at com.mongodb.internal.connection.DefaultConnectionPool.lambda$getAsync$0(DefaultConnectionPool.java:147) ~[main/:na]
```

However, the problem goes away if I actually pass `SingleResultCallback` instead of `groovy.lang.Closure` here.